src/pages-and-resources/discussions/app-config-form/apps/shared/DivisionByGroupFields.jsx

```diff
@@ -1,61 +1,142 @@
-import React from 'react';
+import React, { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
 import PropTypes from 'prop-types';
 import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
 import { Form, TransitionReplace } from '@edx/paragon';
+import { FieldArray, useFormikContext } from 'formik';
 import FormSwitchGroup from '../../../../../generic/FormSwitchGroup';
 import messages from './messages';
 import AppConfigFormDivider from './AppConfigFormDivider';
+import { updateDividedDiscussionsIds } from '../../../data/slice';
+import { updateModel } from '../../../../../generic/model-store';
 
 function DivisionByGroupFields({
-  onBlur,
-  onChange,
-  intl,
-  values,
+  onBlur, onChange, intl, appConfig,
 }) {
+  const dispatch = useDispatch();
+  const { setFieldValue } = useFormikContext();
+  const {
+    dividedDiscussionsIds,
+    discussionTopics,
+    divideByCohorts,
+    divideCourseTopics,
+  } = appConfig;
+
+  useEffect(() => {
+    dispatch(updateDividedDiscussionsIds({ dividedDiscussionsIds }));
+  }, [dividedDiscussionsIds]);
+
+  useEffect(() => {
+    const discussionTopicIds = discussionTopics.map(
+      (topic) => topic.id,
+    );
+    const divideCourseTopicsSwitchOff = (
+      discussionTopicIds.length === dividedDiscussionsIds.length
+      && discussionTopicIds.every((topicId) => dividedDiscussionsIds.includes(topicId))
```

Comment:
I think this check is not required. This should be handled by the API and we should expect the data to be correct

Reply:
> @awais-ansari I have one general question. If we add a new discussion topic and do not submit the form. Will that topic be available to be divided?

yes it will available.

Reply:
@awais-ansari can we cover this in a test case?

Reply:
This case is already cover in test cases.